pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -1166,8 +1189,12 @@ private void createDerivedColumnForwardIndexWithoutDictionary(String column, Fie
             case BYTES:
               forwardIndexCreator.putBytes((byte[]) outputValue);
               break;
+            case MAP:
+              forwardIndexCreator.add(outputValue, -1);
+              break;
             default:
-              throw new IllegalStateException();
+              throw new IllegalStateException(
+                  "Unsupported data type: " + fieldSpec.getDataType() + ", for value: " + outputValue);
```

Comment:
nit: The value is probably not relevant here

Reply:
👍